components/infobox/wikis/dota2/infobox_person_player_custom.lua

```diff
@@ -0,0 +1,271 @@
+	player.createBottomContent = CustomPlayer.createBottomContent
+	player.getCategories = CustomPlayer.getCategories
+	player.createWidgetInjector = CustomPlayer.createWidgetInjector
+
+	return player:createInfobox(frame)
+end
+
+function CustomInjector:parse(id, widgets)
+	if id == 'status' then
+		local statusContents = CustomPlayer._getStatusContents()
+
+		local yearsActive = _args.years_active
+		if String.isEmpty(yearsActive) then
+			yearsActive = YearsActive.display({player = _base_page_name})
+		else
+			yearsActive = Page.makeInternalLink({onlyIfExists = true}, yearsActive)
+		end
+
+		local yearsActiveOrg = _args.years_active_manage
+		if not String.isEmpty(yearsActiveOrg) then
+			yearsActiveOrg = Page.makeInternalLink({onlyIfExists = true}, yearsActiveOrg)
+		end
+
+		return {
+			Cell{name = 'Status', content = statusContents},
+			Cell{name = 'Years Active (Player)', content = {yearsActive}},
+			Cell{name = 'Years Active (Org)', content = {yearsActiveOrg}},
+			Cell{name = 'Years Active (Coach)', content = {_args.years_active_coach}},
+			Cell{name = 'Years Active (Analyst)', content = {_args.years_active_analyst}},
+			Cell{name = 'Years Active (Talent)', content = {_args.years_active_talent}},
+		}
+	elseif id == 'history' then
+		if not String.isEmpty(_args.history_iwo) then
+			table.insert(widgets, Title{name = '[[Intel World Open|Intel World Open]] History'})
+			table.insert(widgets, Center{content = {_args.history_iwo}})
+		end
+		if not String.isEmpty(_args.history_gfinity) then
+			table.insert(widgets, Title{name = '[[Gfinity/Elite_Series|Gfinity Elite Series]] History'})
+			table.insert(widgets, Center{content = {_args.history_gfinity}})
+		end
+		if not String.isEmpty(_args.history_odl) then
+			table.insert(widgets, Title{name = '[[Oceania Draft League|Oceania Draft League]] History'})
+			table.insert(widgets, Center{content = {_args.history_odl}})
+		end
+	elseif id == 'role' then
+		return {
+			Cell{name = 'Current Role', content = {
+					CustomPlayer._createRole('role', _args.role),
+					CustomPlayer._createRole('role2', _args.role2)
+				}},
+		}
+	elseif id == 'nationality' then
+		return {
+			Cell{name = 'Location', content = {_args.location}},
+			Cell{name = 'Nationality', content = CustomPlayer._createLocations()}
+		}
+	end
+	return widgets
+end
+
+function CustomInjector:addCustomCells(widgets)
+	table.insert(widgets,
+		Builder{
+			builder = function()
+				local heroes = Player:getAllArgsForBase(_args, 'hero')
+				local icons = Array.map(heroes,
+					function(h, _)
+						return HeroIcon._getImage{hero = h, size = _SIZE_HERO}
+					end
+				)
+				return {
+					Cell{
+						name = 'Signature Hero',
+						content = {
+							table.concat(icons, '&nbsp;')
+						}
+					}
+				}
+			end
+		})
+	return widgets
+end
+
+function CustomPlayer:createWidgetInjector()
+	return CustomInjector()
+end
+
+function CustomPlayer:makeAbbr(title, text)
```

Comment:
if #1005 goes through could use that instead

Reply:
Let's leave that for now